client/scripts/views/pages/delegate.ts

```diff
@@ -0,0 +1,170 @@
+      // ]),
+    ]);
+  }
+};
+
+interface IDelegateForm {
+  address: string,
+  amount: number,
+}
+
+const DelegateForm: m.Component<{}, { form: IDelegateForm, loading: boolean, currentDelegate: string, }> = {
+  oninit: (vnode) => {
+    vnode.state.form = {
+      address: '',
+      amount: null,
+    };
+    vnode.state.loading = false;
+    getDelegate(vnode);
+  },
+  view: (vnode) => {
+    const { form, loading } = vnode.state;
+    return [
+      m(DelegateStats, {
+        currentDelegate: vnode.state.currentDelegate,
+      }),
+      m(Form, { class: 'DelegateForm' }, [
+        m(Grid, [
+          m(Col, [
+            m('h2', 'Set up your delegation:'),
+            m(FormGroup, [
+              m(FormLabel, 'Your delegate:'),
+              m(Input, {
+                name: 'address',
+                placeholder: 'Paste address you want to delegate to',
+                oninput: (e) => {
+                  const result = (e.target as any).value;
+                  vnode.state.form.address = result;
+                  m.redraw();
+                }
+              }),
+              m(FormLabel, 'Amount of MPOND to delegate:'),
+              m(Input, {
+                name: 'amount',
+                placeholder: '10000',
+                defaultValue: '',
+                oninput: (e) => {
+                  const result = (e.target as any).value;
+                  vnode.state.form.amount = result;
+                  m.redraw();
+                }
+              }),
+            ]),
+            m(FormGroup, [
+              m(Button, {
+                disabled: form.address === '' || loading,
+                intent: 'primary',
+                rounded: true,
+                label: 'Delegate!',
+                onclick: async (e) => {
+                  e.preventDefault();
+                  vnode.state.loading = true;
+                  if ((app.chain as Marlin).apiInitialized) {
+                    await (app.chain as Marlin).marlinAccounts.senderSetDelegate(vnode.state.form.address, vnode.state.form.amount)
+                      .then(async () => {
+                        notifySuccess(`Sent transaction to delegate to ${vnode.state.form.address}`);
+                        await getDelegate(vnode);
+                        m.redraw();
+                      })
+                      .catch((err) => { notifyError(`${err.message}`); });
+                  }
+                  vnode.state.loading = false;
+                  m.redraw();
+                },
+                type: 'submit',
+              }),
+            ]),
+          ]),
+        ]),
+      ]),
+    ];
+  }
+};
+
+const DelegatePage: m.Component<{}> = {
+  view: (vnode) => {
+    if (!app.chain || !app.chain.loaded) {
+      // chain load failed
+      if (app.chain && app.chain.failed) {
+        return m(PageNotFound, {
+          title: 'Wrong Ethereum Provider Network!',
+          message: 'Change Metamask to point to Ropsten Testnet',
```

Comment:
This + 2 other messages need to be switched to Mainnet if you're on the mainnet version.